books/tests/test_views.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+from django.test import RequestFactory
+from django.contrib.auth.models import AnonymousUser
+from mixer.backend.django import mixer
+
+pytestmark = pytest.mark.django_db
+
+from .. import views
+
+class TestHomePageView:
+    def test_anonymous(self):
+        req = RequestFactory().get('/')
+        resp = views.HomePageView.as_view()(req)
+        assert resp.status_code == 200, 'Should be callable by anyone'
+
+class TestBookNewView:
+    def test_anonymous(self):
+        req = RequestFactory().get('/')
+        req.user = AnonymousUser()
+        resp = views.BookNewView.as_view()(req)
+        assert 'login' in resp.url, 'Should not allow access to anonymous'
+
+# did not manage to import CustomUser or any other
```

Comment:
To import your user model or model name you should use:
```python
from django.contrib.auth import get_user_model

CustomUser = get_user_model()
```
This is a django util that will import the correct model for you and make it available.

If you want to refer to the *name* of the model, then you can also simply import it from the settings:

```python
from django.conf import settings
print(settings.AUTH_USER_MODEL)  # `users.CustomUser`
```

Finally, specifically for use in tests, pytest again has a fixture for this, called [django_user_model]( https://pytest-django.readthedocs.io/en/latest/helpers.html#django-user-model) which you should be able to use without any other imports 😄